### PR TITLE
Add script to create Post Galleries from custom blocks.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request creates "Post Gallery" blocks for each of these older campaigns (the preferred workflow in our new multiple-action world) instead of linking one monolithic `customBlock`.

### How should this be reviewed?

This is my first script! @mendelB, please be kind! 🙈

### Any background context you want to provide?

I avoided using `attempt` since I think it'll be less stressful if the script errors out when it runs into a problem so the operator (me) can hop in, sort out the issue, and then continue along (rather than running it and then having a bunch of things to follow up on).

### Relevant tickets

References [Pivotal #171002750](https://www.pivotaltracker.com/story/show/171002750). Unblocks #1900.

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        continue;
+      }
+
+      // Create a new Post Gallery entity for this action.
+      const galleryTitle = `${getField(campaign, 'internalTitle')} Gallery`;
+      const gallery = await environment.createEntry(
+        'postGallery',
+        withFields({
+          internalTitle: galleryTitle,
+          actionIds: [String(photoAction.id)],
```
Comment (mendelB) [inline contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: I doubt this is a very big deal - if at all applicable to these archaic campaigns - but maybe it's worth pulling _all_ the `photoAction`s above, and then passing _all_ action IDs to the gallery for parity since if previously we were using the campaign ID, the galleries _would_ show the posts from all actions?

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        `→ Created "${galleryTitle}" (${gallery.sys.id}, action #${photoAction.id}).`,
+      );
+
+      // Replace the link to the custom block with our new post gallery:
+      page.fields.blocks[LOCALE] = reject(
+        page.fields.blocks[LOCALE],
+        block => block.sys.id === customBlock.sys.id,
+      );
+
+      page.fields.blocks[LOCALE].push(linkReference(gallery.sys.id));
```
Comment (mendelB) [inline contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Do we need to ensure we replace this within the same array index as the gallery block was previously? or are they always at the end of the page?

File: contentful/management-api-scripts/contentManagementClient.js [hunk 3]
```diff
@@ -17,9 +17,11 @@ async function initContentManagementClient(callback) {
 
   if (!spaceId || !accessToken) {
     console.log(
-      'Please provide the space-id and access token arguments in the following format:',
+      'Please provide the space, environment & access token in the following format:',
+    );
+    console.log(
+      '--space-id [id] --access-token [token] --environment-id [master|qa|dev]',
```
Comment (mendelB) [inline contentful/management-api-scripts/contentManagementClient.js]: 🙌

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  linkReference,
+  constants,
+  sleep,
+} = require('./helpers');
+
+const { LOCALE } = constants;
+
+// @TODO: Support using the appropriate Rogue environment?
+const ROGUE_API = 'https://activity-qa.dosomething.org/api/v3';
+const logger = createLogger('replace_staff_linked_entries_with_person');
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Oops! 🍝

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+contentManagementClient.init(async environment => {
+  const blocks = await environment.getEntries({
+    content_type: 'customBlock',
+    'fields.type': 'gallery',
+  });
+
+  for (let i = 0; i < blocks.items.length; i += 1) {
+    const customBlock = blocks.items[i];
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Oh that could read a little nicer, yeah. I had also tried using [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of), but changed back to a simple `for` since it was causing async logic to get jumbled (since that ones uses iterators behind the scenes).

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      await sleep(500);
+
+      // Get the campaign ID for this action page:
+      const campaignQuery = await environment.getEntries({
+        links_to_entry: page.sys.id,
+        limit: 1,
+      });
+
+      const campaign = campaignQuery.items[0];
+      const campaignId = campaign.fields.legacyCampaignId[LOCALE];
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Oh we totally could!

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  withFields,
+  linkReference,
+  constants,
+  sleep,
+} = require('./helpers');
+
+const { LOCALE } = constants;
+
+// @TODO: Support using the appropriate Rogue environment?
+const ROGUE_API = 'https://activity-qa.dosomething.org/api/v3';
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Oh, that's a really nice idea. I think we'd have to use `dotenv` to parse those (and could consider doing the same thing for `CONTENTFUL_SPACE_ID`, etc.)

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+      links_to_entry: customBlock.sys.id,
+    });
+
+    const internalTitle = getField(customBlock, 'internalTitle');
+    logger.info(
+      `\n\nProcessing ${linkedEntries.items.length} links to "${internalTitle}".`,
+    );
+
+    for (let j = 0; j < linkedEntries.items.length; j += 1) {
+      const page = linkedEntries.items[j];
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: I did a quick check of this on QA & it is indeed only pages! 😌

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 1 above)
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Mmmm, I was operating under the assumption that these were all old database campaigns with a single `action` but it's probably safer to grab 'em all. And actually, would we want to include `text` actions in that case too...?

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 2 above)
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Ahh, yeah I was making that assumption. 🙈 Maybe instead we should find the index of that one that we're rejecting above & instead do a lil' `splice` to add it's replacement.

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (mendelB) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Oh 🤦‍♂right it's an array! I _feel_ like I must have used this before though for scripts like these, maybe it was [`for await...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of)?

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 1 above)
Comment (mendelB) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Ooh good point, I guess it can't hurt right? We'd only be consistent at worst...

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Yeah, that worked! The problem I ran into is that then whenever we blocked on an `await`, it'd move on to another loop iteration that wasn't blocked (which is a total performance win, but makes script output a little hard to follow).

[the remark this answers is not in the document]
Comment (mendelB) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Ohhhh I see! well old school is good school 😎

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Updated in abb0c0f.

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Updated in a759a4d! I updated the script to log the chosen environment & pause for a bit before starting things up so we don't accidentally start running against the wrong space:

Screen Shot 2020-02-05 at 11 44 12 AM

File: contentful/management-api-scripts/2020_01_30_create_post_galleries.js (hunk 2 above)
Comment (DFurnes) [reply contentful/management-api-scripts/2020_01_30_create_post_galleries.js]: Done in 4696770!
